### PR TITLE
feat(tauri.js) improve api module with type defs for each API function

### PR DESCRIPTION
**Pull request by lucasfernog**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch and _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: cli/tauri.js/api/event.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import tauri from './tauri'
+
+/**
+ * listen to an event from the backend
+ */
+function listen (event, handler) {
+  tauri.listen(event, handler)
+}
+
+function emit (event, payload) {
```
Comment (nothingismagick) [inline cli/tauri.js/api/event.js]: Wish there was JSDoc here.

File: cli/tauri.js/api/fs.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ * @param {string} filePath path to the file
+ * @return {Promise<int[]>}
+ */
+function readBinaryFile (filePath) {
+  return tauri.readBinaryFile(filePath)
+}
+
+/**
+ * writes a text file
+ *
```
Comment (nothingismagick) [inline cli/tauri.js/api/fs.js]: these JSDocs are wrong.

**Comment by lucasfernog** [comment]
this was supposed to be a JS PR but I'm thinking about adding some API functions and more refactoring too.

File: cli/tauri.js/api/window.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import tauri from './tauri'
+
+function setTitle (title) {
+  tauri.setTitle(title)
+}
+
+function open (url) {
+  tauri.open(url)
+}
+
```
Comment (nothingismagick) [inline cli/tauri.js/api/window.js]: we should add the ability to change window size, transparency (maybe in the near future!)

File: cli/tauri.js/templates/tauri.js [hunk 4]
```diff
@@ -398,26 +398,6 @@ window.tauri = {
         <% } %>
   },
 
-bridge: function bridge(command, payload) {
-    <% if (tauri.whitelist.bridge === true || tauri.whitelist.all === true) { %>
```
Comment (nothingismagick) [inline cli/tauri.js/templates/tauri.js]: good call - should've been removed LONG ago

File: cli/tauri.js/api/window.js (hunk 3 above)
Comment (nothingismagick) [reply cli/tauri.js/api/window.js]: @lucasfernog - what do you think about this?

Comment (lucasfernog) [reply cli/tauri.js/api/window.js]: I'll make it on a second PR. This one is too big already, but I want to make more additions.

Comment (lucasfernog) [reply cli/tauri.js/api/window.js]: window size change is only supported by zserge, so i'll do it when we get the chance.

**Pull request merged**